v_3_0_0/master_template.go

```diff
@@ -360,210 +360,143 @@ write_files:
               hostPath:
                 path: /etc/kubernetes/ssl/etcd
 {{ end -}}
-- path: /srv/kubedns-cm.yaml
+- path: /srv/coredns.yaml
   owner: root
   permissions: 0644
   content: |
     apiVersion: v1
-    kind: ConfigMap
+    kind: ServiceAccount
     metadata:
-      name: kube-dns
+      name: coredns
       namespace: kube-system
+    ---
+    apiVersion: rbac.authorization.k8s.io/v1beta1
+    kind: ClusterRole
+    metadata:
       labels:
-        addonmanager.kubernetes.io/mode: EnsureExists
-- path: /srv/kubedns-sa.yaml
-  owner: root
-  permissions: 0644
-  content: |
+        kubernetes.io/bootstrapping: rbac-defaults
+      name: system:coredns
+    rules:
+    - apiGroups:
+      - ""
+      resources:
+      - endpoints
+      - services
+      - pods
+      - namespaces
+      verbs:
+      - list
+      - watch
+    ---
+    apiVersion: rbac.authorization.k8s.io/v1beta1
+    kind: ClusterRoleBinding
+    metadata:
+      annotations:
+        rbac.authorization.kubernetes.io/autoupdate: "true"
+      labels:
+        kubernetes.io/bootstrapping: rbac-defaults
+      name: system:coredns
+    roleRef:
+      apiGroup: rbac.authorization.k8s.io
+      kind: ClusterRole
+      name: system:coredns
+    subjects:
+    - kind: ServiceAccount
+      name: coredns
+      namespace: kube-system
+    ---
     apiVersion: v1
-    kind: ServiceAccount
+    kind: ConfigMap
     metadata:
-      name: kube-dns
+      name: coredns
       namespace: kube-system
-      labels:
-        kubernetes.io/cluster-service: "true"
-        addonmanager.kubernetes.io/mode: Reconcile
-- path: /srv/kubedns-dep.yaml
-  owner: root
-  permissions: 0644
-  content: |
-    apiVersion:  extensions/v1beta1
+    data:
+      Corefile: |
+        .:53 {
+            errors
+            health
+            kubernetes {{.Cluster.Kubernetes.Domain}} {{.Cluster.Kubernetes.API.ClusterIPRange}} {{.Cluster.Calico.Subnet}}/{{.Cluster.Calico.CIDR}} {
+              pods insecure
+              upstream /etc/resolv.conf
+            }
+            prometheus :9153
+            proxy . /etc/resolv.conf
+            cache 30
+        }
+    ---
+    apiVersion: extensions/v1beta1
     kind: Deployment
     metadata:
-      name: kube-dns
+      name: coredns
       namespace: kube-system
       labels:
-        k8s-app: kube-dns
-        kubernetes.io/cluster-service: "true"
-        addonmanager.kubernetes.io/mode: Reconcile
+        k8s-app: coredns
+        kubernetes.io/name: "CoreDNS"
     spec:
+      replicas: 2
```

Comment:
why only 2 and not 3? or is this how it's done upstream?

Reply:
this it pure upstream manifest https://github.com/coredns/deployment/blob/master/kubernetes/coredns.yaml.sed

Reply:
nice, all good then, we will need to see how it scales with lots of requests, but I heard CoreDNS is quite performant and nice